packages/keystone/plugins/addressService.js

```diff
@@ -0,0 +1,114 @@
+const conf = require('@open-condo/config')
+const { ADDRESS_META_FIELD } = require('@open-condo/keystone/plugins/utils/addressMetaDefinition')
+const { composeResolveInputHook } = require('@open-condo/keystone/plugins/utils')
+const { plugin } = require('@open-condo/keystone/plugins/utils/typing')
+const { Text } = require('@keystonejs/fields')
+const {
+    createInstance: createAddressServiceClientInstance,
+    createTestInstance: createTestAddressServiceClientInstance,
+} = require('@open-condo/keystone/plugins/utils/address-service-client')
+const get = require('lodash/get')
+
+const readOnlyAccess = {
+    read: true,
+    create: false,
+    update: false,
+    delete: false,
+}
+
+const getSourceFieldName = (mainFieldName) => `${mainFieldName}Source`
+const getKeyFieldName = (mainFieldName) => `${mainFieldName}Key`
+const getMetaFieldName = (mainFieldName) => `${mainFieldName}Meta`
+
+const getFieldsToAdd = (mainField, fieldsHooks) => ({
+    [mainField]: {
+        schemaDoc: 'Normalized address',
+        type: Text,
+        isRequired: true,
+        hooks: {
+            ...(fieldsHooks[mainField] || {}),
+        },
+    },
+    [getSourceFieldName(mainField)]: {
+        type: Text,
+        schemaDoc: 'The origin of the address (some string which may looks like real address or some id)',
+        access: readOnlyAccess,
+        hooks: {
+            ...(fieldsHooks[getSourceFieldName(mainField)] || {}),
+        },
+    },
+    [getKeyFieldName(mainField)]: {
+        type: Text,
+        schemaDoc: 'The unique key of the address',
+        access: readOnlyAccess,
+        hooks: {
+            ...(fieldsHooks[getKeyFieldName(mainField)] || {}),
+        },
+    },
+    [getMetaFieldName(mainField)]: ADDRESS_META_FIELD,
+})
+
+/**
+ * Plugin to extend address field with address service
+ * @param {string?} addressFieldName
+ * @param {Object} fieldsHooks
+ */
+const addressService = (addressFieldName = 'address', fieldsHooks = {}) => plugin(({
```

Comment:
use the same argument pattern as in others plugins. Just one argument with all options

Reply:
Done